### DISCUSSION TITLE
install 2.7.1 on Ubuntu Build Failed

### DISCUSSION THREAD

**Discussion by daBee** [rbenv/ruby-build#2172] [Build failures] [answered]
```
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
```

`rbenv install 2.7.1`

That version is the most recent stable version. Here is the last 10 lines:

```
Last 10 log lines:
make[2]: Leaving directory '/tmp/ruby-build.20230329220617.26255.cr8ssr/ruby-2.7.1/ext/openssl'
make[1]: *** [exts.mk:251: ext/openssl/all] Error 2
make[1]: *** Waiting for unfinished jobs....
installing default date_core libraries
linking shared-object bigdecimal.so
make[2]: Leaving directory '/tmp/ruby-build.20230329220617.26255.cr8ssr/ruby-2.7.1/ext/bigdecimal'
linking shared-object date_core.so
make[2]: Leaving directory '/tmp/ruby-build.20230329220617.26255.cr8ssr/ruby-2.7.1/ext/date'
make[1]: Leaving directory '/tmp/ruby-build.20230329220617.26255.cr8ssr/ruby-2.7.1'
make: *** [uncommon.mk:295: build-ext] Error 2
```

Info about `ruby-build`:
```
Package: ruby-build
Version: 20200401-1build1
Priority: optional
Section: universe/ruby
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian Ruby Extras Maintainers <pkg-ruby-extras-maintainers@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 577 kB
Depends: build-essential, curl, libreadline6-dev, zlib1g-dev
Recommends: libsqlite3-dev, libssl-dev, libxml2-dev, libxslt-dev, rbenv
Suggests: autoconf, automake, bison, git-core, libtool
Homepage: https://github.com/rbenv/ruby-build
Download-Size: 57.0 kB
APT-Manual-Installed: yes
APT-Sources: http://ports.ubuntu.com jammy/universe arm64 Packages
Description: tool to compile and install different versions of Ruby
```

Not sure what to do here.

**Comment by hsbt** [comment, marked as answer]
Unfortunately, you can install Ruby 3.1 and 3.2 with Ubuntu 22.04. Ruby 2.7 and 3.0 didn't support OpenSSL 3.x provided by Ubuntu 22.04.

**Comment by daBee** [comment]
OK I'm jumping the gun a bit here then.  Ruby 3.1 and 3.2 aren't available on this platform.  At least through `rbenv`.

**Reply by mislav** [reply]
The ruby-build version `20200401-1build1` you received from a package repository is three years out of date and thus will not know how to install new Ruby version like Ruby 3.1.

To install Ruby 3.x with ruby-build, you should get a newer version of ruby-build. If you view our README installation instructions, you can find a method of installation that is just `git clone` of the latest version from this GitHub repository.

To install Ruby 2.7 on your system, you might first have to download and compile OpenSSL 1.1, then link your Ruby build to it as described in https://github.com/rbenv/ruby-build/discussions/1940#discussioncomment-2519546. However, a newer version of ruby-build might do that for you automatically when it detects that system `openssl` is too new to be compatible with older Ruby.

**Comment by daBee** [comment]
`git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build`

I'm unclear where that target directory should be.  `~/.rbenv` has two empty directories, neither being `~/.rbenv/plugins`.

**Reply by mislav** [reply]
If you are intending to use the rbenv version switcher after installation, then you should set up rbenv first. After that, `rbenv root` will return the correct location and the `git  clone` you are trying to use will just work. This will be evident by you being able to run `rbenv install --list` and see the latest Ruby versions. (It's fine if the `plugins` directory didn't exist before or was empty.)

If your entire purpose for ruby-build is to just install a Ruby version (and you don't intend to use rbenv), then you don't need to place ruby-build anywhere specific. Just download it anywhere and run `bin/ruby-build`.
